third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -212,18 +197,21 @@ impl Interpreter {
                         // end of execution. `self` should no longer be used afterward
                         // Clean up access control
                         self.access_control
-                            .exit_function(current_frame.function.as_ref())
+                            .exit_function(current_frame.function.definition())
                             .map_err(|e| self.set_location(e))?;
                         return Ok(self.operand_stack.value);
                     }
                 },
                 ExitCode::Call(fh_idx) => {
-                    let func = resolver
+                    let (module, func) = resolver
```

Comment:
This gets somewhat uglier: we hget the module and the function for the next iteration and call frame, so tat we can use a new resolver cc @runtian-zhou